core/scm.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+const Joi = require('joi');
+
+const SCHEMA_USER = Joi.object().keys({
+    id: Joi.string()
+        .required()
+        .label('Unique Identifier')
+        .example('github.com:622065'),
+
+    url: Joi.string()
+        .uri()
+        .required()
+        .label('Link to Profile')
+        .example('https://github.com/stjohnjohnson'),
+
+    name: Joi.string()
+        .required()
+        .label('Display Name')
+        .example('stjohnjohnson'),
+
+    avatar: Joi.string()
+        .uri()
+        .required()
```

Comment:
Pretty much all of them: 
 - https://docs.gitlab.com/ee/customization/libravatar.html
 - https://blog.bitbucket.org/2013/07/16/customize-your-avatar-in-bitbucket-no-gravatar-needed/